Shared/Flows/Onboarding/OnboardingCoordinator.swift

```diff
@@ -21,40 +21,142 @@ class OnboardingCoordinator: PresentableCoordinator<Void> {
     }
 
     override func start() {
+        self.setInitialOnboardingContent()
         self.handle(deeplink: self.deepLink)
     }
 
     func handle(deeplink: DeepLinkable?) {
         guard let link = deeplink else { return }
+
         self.onboardingVC.reservationId = link.reservationId
         self.onboardingVC.passId = link.passId
         self.onboardingVC.updateUI()
     }
-}
 
-extension OnboardingCoordinator: LaunchActivityHandler {
+    // MARK: - Onboarding Flow Logic
```

Comment:
I moved all of the onboarding flow logic into the coordinator for better separation of concern.